### DISCUSSION TITLE
ebusd & MQTT integration

### DISCUSSION THREAD

**Discussion by FrigerioSergio** [john30/ebusd#544] [Q&A]
Hello to everyone.
I am trying to configure ebusd and MQTT on my raspberry Pi4 running Home Assistant OS.
Hardware interface:  Esera eBus USB

I followed the instruction founded here: https://community.home-assistant.io/t/an-ebusd-add-on/344852

This is the log from ebusd:

_[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
> ebusd --foreground --mqtthost=core-mosquitto --mqttport=1883 --mqttuser=addons --mqttpass=cieZu3vaigh0bah8yi8Ienooth1mi9aes6iz9ur3eethah6HaegahJaigh7naith --scanconfig --mqttjson --pollinterval=6 --mqttint=/etc/ebusd/mqtt-hassio.cfg --mqtttopic=ebusd --device=/dev/ttyUSB0 --log=all:notice
2022-02-25 09:28:57.408 [main notice] ebusd 22.1.v22.1-44-gbb96517 started with auto scan on device /dev/ttyUSB0
2022-02-25 09:28:58.521 [bus notice] bus started with own address 31/36
2022-02-25 09:28:58.527 [mqtt notice] connection established
2022-02-25 09:28:58.532 [bus notice] signal acquired
2022-02-25 09:28:58.692 [bus notice] new master 37, master count 2
2022-02-25 09:28:58.693 [update notice] received unknown BC cmd: 37fe20500b5555020d00000000000000
2022-02-25 09:28:59.210 [update notice] received unknown BC cmd: 37fe20510e1602555555555555555500000000
2022-02-25 09:28:59.416 [update notice] received unknown BC cmd: 37fe2000026197
2022-02-25 09:28:59.496 [bus notice] new master 13, master count 3
2022-02-25 09:28:59.496 [update notice] received unknown BC cmd: 13fe200f056197310100
2022-02-25 09:29:00.058 [bus notice] new master 01, master count 4
2022-02-25 09:29:00.058 [update notice] received unknown BC cmd: 01fe2010057318cf0000
2022-02-25 09:29:00.552 [bus notice] new master 70, master count 5
2022-02-25 09:29:00.584 [update notice] received unknown MS cmd: 703c2000047547c404 / 04030e0000
2022-02-25 09:29:00.692 [bus notice] new master 7f, master count 6
2022-02-25 09:29:00.692 [update notice] received unknown MS cmd: 3784200008d140cc4bc89ccc9a / 050f01080000_

I am stucked here:
![image](https://user-images.githubusercontent.com/22483463/155681469-55c02ad3-b6fe-438b-b8d5-2e8e4e1b3861.png)

As far as I understood, addon should automatically find the sensor but nothing happen.

**How to get data into Home Assistant** 
MQTT discovery is now automatically configured. When you start the add-on the global readings (uptime, signal status etc.) will be added to Home Assistant automatically under a device named ebusd. After around 5 minutes any readings that are being polled, either via an MQTT request or by editing your config files, will be added to Home Assistant automatically. The device name will be ebusd {circuit} - e.g. “ebusd bai”.

The mqttvar option can be used to inject MQTT variables or the [mqtt-hassio.cfg 3](https://github.com/john30/ebusd/blob/5ec5c3daa44502b3bb5a6162797cc47dfe4c5f23/contrib/etc/ebusd/mqtt-hassio.cfg) can be edited and saved in your /config folder.

For more info please see the eBUSd docs - [MQTT Integration}([https://github.com/john30/ebusd/wiki/MQTT-integration 6](https://github.com/john30/ebusd/wiki/MQTT-integration))

I did not find anything similar on my Home Assistant config folder.....

Any help will be appreciated.

**Comment by Koky05** [comment]
It is not in your HA config folder. But in ebusd-deamon folder. I map it like (I use Synology NAS and docker-compose):
volumes:
      - "/volume1/docker/ebusd/config:/etc/ebusd"

Then I can find that configuration file in folder /volume1/docker/ebusd/config/

**Reply by FrigerioSergio** [reply]
Hello, thanks for the answer. i don't know how to reach ebusd-daemon folder.....
Could you please help me?

**Reply by Koky05** [reply]
If I look into your first topic try direct folder:
cd /etc/ebusd/

**Reply by LukasGrebe** [reply]
The `/config` folder from HA is mapped to `/config` in the ebusd add-on. if you create eg an MQTT integration config unter `/config/ebusd/myCustomMQTT.cfg` you can have ebusd use it by setting the mqtt cfg configuration file in the HA add-on to `/config/ebusd/myCustomMQTT.cfg` wich is the same as setting command line options to ` --mqttint=/config/ebusd/myCustomMQTT.cfg`

but as @john30 has mentioned, most Users don't need to copy any configuration as it is only necessary if a local modification to the defaults are needed.

**Comment by FrigerioSergio** [comment]
![image](https://user-images.githubusercontent.com/22483463/155695372-4e348e5a-1877-4c83-8ce3-e0ebdc4605da.png)


Nothing....

**Reply by Koky05** [reply]
OK, you need to link configuration:
https://github.com/john30/ebusd-configuration#installation-on-up-to-date-ebusd-20-or-higher

**Reply by FrigerioSergio** [reply]
Seems that it's too hard for me end my knowledge
 I've run the 3 command line. executed without error but nothing happen.
Thanks for your help.

**Reply by Koky05** [reply]
Type all three commands here. And we will found solution for you.
Or try this instructions:
https://github.com/john30/ebusd/issues/135#issuecomment-348799109

**Comment by john30** [comment]
most users don't have to clone anything and the config is automatically downloaded from the web service by default.
do you see decoded messages in the terminal running ebusd and does ebusd log some scan success entries?

**Reply by FrigerioSergio** [reply]
Hello. Not sure if I have understood properly your question.

I am trying to connect my Home Assistant to an Elco/Ariston heating system. I know that it's a little bit different form the Vailant one, but I also read that someone has been able to "decrypt" the ariston Ebus message:

https://it.emcelettronica.com/energybus-un-bus-per-gli-impianti-di-climatizzazione
https://ebus-wiki.org/doku.php/ebus/ebuskonverter

The idea was to at least been abel to read the messages and maybe later write also some command...

**Comment by john30** [comment]
I see. in this case you'd have to get these configs and configure ebusd to use them. in a docker container you'd mount a local directory into it or create a dedicated volume for it

**Reply by FrigerioSergio** [reply]
Ok. Finally, starting from this job: https://github.com/komw/ariston-bus-bridgenet-ebusd  (many thanks) and your help now I can read some sensors from my heating system.

**Reply by wrongisthenewright** [reply]
Ciao Sergio,
for Ariston/BusBridgenet(R) related devices you can also try my config file, you can find it at:

https://github.com/wrongisthenewright/ebusd-configuration-ariston-bridgenet

**Reply by FrigerioSergio** [reply]
Ciao wrong.... (don't know your name....),
I have already looked at your file (just received a couple of days ago the new ebus adapter).
I got a lot of good value but also a not known ones.
Question, did you speak italian?

Thanks,

Sergio

**Reply by wrongisthenewright** [reply]
Ciao Sergio, yes I still can speak italian:)
